src/main/java/map/BattleMap.java

```diff
@@ -17,11 +17,60 @@ public BattleMap(int size) {
     this.map.addAll(Collections.nCopies(size * size, 0));
   }
 
+  public List<Integer> getMap() {
+    return map;
```

Comment:
Here we still return the "actual" map, and it still can be mutated. What we need is to return a copy of the actual map.